### DISCUSSION TITLE
/devhack-rename-file-vscode-extension/

### DISCUSSION THREAD

**Discussion by comment-migrator** [estruyf/eliostruyf-comments#577] [Comments]
#DevHack: How rename a file from an extension
        
[https://www.eliostruyf.com/devhack-rename-file-vscode-extension/](https://www.eliostruyf.com/devhack-rename-file-vscode-extension/)

**Comment by zatura328** [comment]
Nice solution i had to use fs to check if fileExist before do yo know if there is a vscode.workspace.fs built in function for FileExist. I dont want to use readFile i found it to overkill.

**Reply by zatura328** [reply]
Sorry for the error in the verb i find it to overkill.

**Reply by estruyf** [reply]
There is no fileExists method, but you can use stats. Here is an example: https://github.com/estruyf/vscode-demo-time/blob/dev/src/utils/fileExists.ts

**Reply by zatura328** [reply]
Thanks !!! It's less memory consuming than using the both fs and vscode.workspace.fs.
